token-lending/program/src/state/reserve.rs

```diff
@@ -606,9 +653,105 @@ impl IsInitialized for Reserve {
     }
 }
 
-const RESERVE_LEN: usize = 602;
+// @TODO: adjust padding. what's a reasonable number?
+const RESERVE_LEN: usize = 610; // 1 + 8 + 1 + 32 + 32 + 1 + 32 + 32 + (4 + 32) + 16 + 8 + 8 + 16 + 32 + 32 + 8 + 1 + 1 + 1 + 1 + 1 + 1 + 8 + 1 + 300
 impl Pack for Reserve {
-    const LEN: usize = 602;
+    const LEN: usize = RESERVE_LEN;
+
+    fn pack_into_slice(&self, output: &mut [u8]) {
+        let output = array_mut_ref![output, 0, RESERVE_LEN];
+        #[allow(clippy::ptr_offset_with_cast)]
+        let (
+            version,
+            last_update_slot,
+            last_update_stale,
```

Comment:
`last_update_slot` and `last_update_stale` are both correct here -- the nested `LastUpdate` struct has both of these, and uses them differently. Because reserves and obligations can become stale even in the current slot if deposits/borrows/values/interest change, there needs to be a way to mark them as stale so that later instructions don't trust their values.

Reply:
Ah, my mistake, thanks for clarifying!